java/sage/Seeker.java

```diff
@@ -1241,6 +1241,17 @@ else if (autoImportAddedNetPaths.size() > 0)
       writeArchiveDirsProps();
     }
 
+    // Even though this is the only method that manipulates this variable, it is run on a thread
```

Comment:
The fact that a different thread could be working with the data on each pass (because we are using Pooler) means that we thread caching is possible. Your confusion makes me think that maybe my logic is backwards, but I believe because it's a local variable and not parameter, it's not actually being synchronized by Pooler when it kicks off the runnable on one of the pooled threads. Just because one thread at any time works on data doesn't mean it's thread safe. If each thread is a different thread, they can potential see the data differently because they are pulling the variable in through the local cache (thread caching), not main memory.